### ISSUE TITLE
Remove "AND provider='email'" from SQL when login

### ISSUE THREAD

**Issue by casertap** [lynndylanhurley/devise_token_auth#266]
Hi,
I want my users to be able to login with facebook and email to the same account.
The problem with devise_token_auth is that it check the provider field on login:

```
  SELECT  `users`.* FROM `users` WHERE (BINARY email = 'aaaaa@gmail.com' AND provider='email')  ORDER BY `users`.`id` ASC LIMIT 1 
```

Is it possible to remove the "AND provider='email'" from the conditional? Or do you have another way to resolve my problem.

**Comment by nicolas-besnard** [comment]
If your users want to connect via Facebook, you use the Facebook Connect button, otherwise, they use their emails.

**Comment by casertap** [comment]
ok let say my user connected with facebook first than the provider field of my user is set to 'facebook'.
Now my user logout and log using email and password. The sql
    SELECT  `users`.* FROM `users` WHERE (BINARY email = 'aaaaa@gmail.com' AND provider='email')  ORDER BY `users`.`id` ASC LIMIT 1 

does not find any user because the provider of my user was set ta 'facebook'

**Comment by nicolas-besnard** [comment]
I think your user need to connected via email first, then you can link his Facebook account with it.

You might need additional fields such as "facebook_id" and you'll need to change the query made on login.

**Comment by casertap** [comment]
Hi nicolas, thanks for your answers.
I was testing what you propose has a solution and I am wondering how I can change the query that is done at login because this query is done in the devise_token_auth gem.
The solution to my problem will be to not care about the provider field at all.

**Comment by nicolas-besnard** [comment]
As the `SessionsController` is located in `app/controller/devise_token_auth`, you can create a new controller at this place and it'll override the one provided by this gem.

You can also create a new controller in any places you want and specify in your `routes.rb` the right controller to use.

**Comment by casertap** [comment]
Thanks for the quick answer. I let you know if I have problems.

**Comment by casertap** [comment]
Actually, it is very difficult for me to progress because of the issue I get when connecting to facebook: https://github.com/lynndylanhurley/devise_token_auth/issues/268

**Comment by MaicolBen** [comment]
Resolved in #964
